diff --git a/src/mem_to_banks.sv b/src/mem_to_banks.sv
index ca215bd..73addb6 100644
--- a/src/mem_to_banks.sv
+++ b/src/mem_to_banks.sv
@@ -115,7 +115,7 @@ module mem_to_banks #(
     assign bank_req[i].atop  = atop_i;
     assign bank_req[i].we    = we_i;
     stream_fifo #(
-      .FALL_THROUGH ( 1'b1         ),
+      .FALL_THROUGH ( 1'b0         ),
       .DATA_WIDTH   ( $bits(req_t) ),
       .DEPTH        ( FifoDepth    ),
       .T            ( req_t        )
